File: apps/web/src/features/ci/api/ci-workspace-readiness.ts
export interface CiWorkspaceReadinessContract {
  contract_version: "ci_workspace_readiness_v3";
  product_id: "commercial_and_industrial";
  availability: "unavailable" | "evidence_limited";
  active_profile_id: string | null;
  active_profile_label: string | null;
  blockers: Array<{ code: string; message: string }>;
  workspace_areas: Array<{
    workspace_id: string;
    display_label: string;
    description: string;
    availability: "unavailable" | "input_required" | "evidence_limited";
  }>;
}

const expectedWorkspaceIds = [
  "data_qc",
  "tariff_mapping",
  "peak_shaving",
  "kw_kva_pf_evidence",
  "scenario_ranking",
  "report_preview",
] as const;

export const ciWorkspaceReadinessQueryKey = ["ci-workspace-readiness"] as const;

export async function fetchCiWorkspaceReadiness(
  fetcher: typeof fetch = fetch,
): Promise<CiWorkspaceReadinessContract> {
  const response = await fetcher("/api/commercial-industrial/workspace-readiness", {
    headers: { Accept: "application/json" },
  });
  if (!response.ok) {
    throw new Error(`C&I workspace readiness request failed with status ${response.status}.`);
  }
  const payload = (await response.json()) as CiWorkspaceReadinessContract;
  if (payload.contract_version !== "ci_workspace_readiness_v3") {
    throw new Error("C&I workspace readiness returned an unexpected contract version.");
  }
  if (
    payload.product_id !== "commercial_and_industrial" ||
    !["unavailable", "evidence_limited"].includes(payload.availability) ||
    !Array.isArray(payload.blockers) ||
    payload.blockers.length === 0 ||
    !Array.isArray(payload.workspace_areas) ||
    payload.workspace_areas.length !== expectedWorkspaceIds.length ||
    payload.workspace_areas.some(
      (area, index) =>
        area.workspace_id !== expectedWorkspaceIds[index] ||
        !["unavailable", "input_required", "evidence_limited"].includes(area.availability) ||
        !area.display_label ||
        !area.description,
    )
  ) {
    throw new Error("C&I workspace readiness returned an unsafe availability contract.");
  }
  const evidenceAreaIds = new Set(["data_qc", "tariff_mapping", "kw_kva_pf_evidence"]);
  if (
    payload.availability === "evidence_limited" &&
    (!payload.active_profile_id ||
      !payload.active_profile_label ||
      payload.workspace_areas.some((area) =>
        evidenceAreaIds.has(area.workspace_id)
          ? area.availability !== "evidence_limited"
          : area.availability !== "input_required",
      ))
  ) {
    throw new Error("C&I workspace readiness returned an unsafe evidence profile.");
  }
  return payload;
}

File: solar_battery/ci_workspace_readiness.py
from __future__ import annotations

from solar_battery.ci_tariff_analysis import ci_analysis_availability

CI_EVIDENCE_GATE_BLOCKER = {
    "code": "ci_evidence_gate_issue_5",
    "message": (
        "Unavailable while Issue #5 C&I-EVIDENCE-PARKED remains blocked. "
        "No demand-charge, peak-shaving, kVA or power-factor customer-dollar "
        "capability is implied."
    ),
}

_WORKSPACE_AREAS = (
    ("data_qc", "C&I Data QC", "Interval-data suitability and quality evidence."),
    (
        "tariff_mapping",
        "C&I Tariff Mapping",
        "Tariff structure and demand-window evidence.",
    ),
    (
        "peak_shaving",
        "Peak Shaving",
        "Physical dispatch review after explicit battery inputs are provided.",
    ),
    (
        "kw_kva_pf_evidence",
        "kW/kVA/PF Evidence",
        "Power, apparent-power and power-factor evidence.",
    ),
    (
        "scenario_ranking",
        "C&I Scenario Ranking",
        "Physical review ordering after two to four scenarios are run.",
    ),
    (
        "report_preview",
        "C&I Report Preview",
        "In-app physical evidence preview; no downloadable customer report.",
    ),
)


def ci_workspace_readiness_contract() -> dict[str, object]:
    """Return Python-owned C&I readiness, limited by local evidence."""
    supported = ci_analysis_availability()
    evidence_ready = supported["availability"] == "evidence_limited"
    available_ids = {"data_qc", "tariff_mapping", "kw_kva_pf_evidence"}
    return {
        "contract_version": "ci_workspace_readiness_v3",
        "product_id": "commercial_and_industrial",
        "availability": (
            "evidence_limited" if evidence_ready else "unavailable"
        ),
        "active_profile_id": supported["profile_id"],
        "active_profile_label": supported["profile_label"],
        "blockers": [dict(CI_EVIDENCE_GATE_BLOCKER)],
        "workspace_areas": [
            {
                "workspace_id": workspace_id,
                "display_label": display_label,
                "description": description,
                "availability": (
                    "evidence_limited"
                    if evidence_ready and workspace_id in available_ids
                    else "input_required"
                    if evidence_ready
                    else "unavailable"
                ),
            }
            for workspace_id, display_label, description in _WORKSPACE_AREAS
        ],
    }
